orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -707,15 +707,15 @@ IMPLEMENT_GRADIENT_BUILDER(GetSigmoidGradient) {
 
 IMPLEMENT_GRADIENT_BUILDER(GetSoftmaxGradient) {
   return std::vector<NodeDef>{
-      NodeDef(OpDef{"SoftmaxGrad", kMSDomain, 1},
+      NodeDef(OpDef{SrcNodeOpsetVersion() < 13 ? "SoftmaxGrad" : "SoftmaxGrad_13", kMSDomain, 1},
```

Comment:
>"SoftmaxGrad_13" [](http://example.com/codeflow?start=65&length=16)

is versioning of gradient ops going to be handled by appending version numbers to the name?
why do we need the separate op? it's not obvious to me as the kernel registrations point to the same implementation, which also has logic for checking the opset version

Reply:
Yes. I understand that we want to keep the Contrib ops in MSDomain version #1 fixed. So are not versioning the contrib operators, including the gradient operators of ONNX operators. In the kernel implementation we ae checking the opset version of the forward operator.